site/src/api/api.ts

```diff
@@ -1020,3 +1031,26 @@ const getMissingParameters = (
 
   return missingParameters
 }
+
+export const watchBuildLogs = (
+  versionId: string,
+  onMessage: (log: TypesGen.ProvisionerJobLog) => void,
+) => {
+  return new Promise<void>((resolve, reject) => {
+    const proto = location.protocol === "https:" ? "wss:" : "ws:"
+    const socket = new WebSocket(
+      `${proto}//${location.host}/api/v2/templateversions/${versionId}/logs?follow=true`,
+    )
+    socket.binaryType = "blob"
+    socket.addEventListener("message", (event) =>
+      onMessage(JSON.parse(event.data) as TypesGen.ProvisionerJobLog),
+    )
+    socket.addEventListener("error", () => {
+      reject(new Error("Connection for logs failed."))
```

Comment:
Will the socket close automatically after a couple of failures?

Reply:
Good question, let me investigate that.

Reply:
We could maybe make use of a `useEffect` in the component itself to make sure the socket is cleaned up when the user navigates away from the page, even if the logs haven't finished streaming.

Reply:
if this is used on react-query or xstate - they should cleanup those things but I will check. This was already working, I just moved into a function on API to make it easier to mock it.

Reply:
It is not closing the connection!!! This is a thing that I didn't expect.

Reply:
Related to the websocket, I think this is happening for all the other ws connections that we have on the site. I created a ticket to take it separately  https://github.com/coder/coder/issues/6782

Reply:
Thanks for doing that!